ts/sagas/startup/generateCryptoKeyPair.ts

```diff
@@ -93,6 +97,35 @@ function* generateCryptoKeyPair(keyTag: string) {
   }
 }
 
+/**
+ * Get the public key liked the provided keyTag
+ */
+export function* getCryptoPublicKey(keyTag: O.Option<string>) {
+  const emptyKeyInfo: KeyInfo = {
+    keyTag: undefined,
+    publicKey: undefined
+  };
+  try {
+    if (O.isSome(keyTag)) {
+      const publicKey = yield* call(getPublicKey, keyTag.value);
+      const keyInfo: KeyInfo = {
+        keyTag: keyTag.value,
+        publicKey,
+        publicKeyThumbprint: jwkThumbprintByEncoding(
+          publicKey,
+          DEFAULT_LOLLIPOP_HASH_ALGORITHM_CLIENT,
+          "base64url"
+        )
+      };
+      return keyInfo;
+    } else {
+      return emptyKeyInfo;
+    }
+  } catch (e) {
```

Comment:
does this throws? In caso, I'd handle exceptions using `tryCatch` from `Either` or `TaskEithers` instead

Reply:
As above.